[Tests] Refactored parsing of scheme from repo URL, added Collection tests

czechboy0 opened the pull request:
- refactored parsing of scheme from repo URL
- moved Array utils to Collection utils 
- added Collection tests 
- moved all allTests functions into one file (Utilities modules only for now), previously they were in FileTests which was hard to find

gribozavr commented on Sources/Utility/CollectionExtensions.swift:
```diff
@@ -0,0 +1,72 @@
+    }
+}
+
+//FIXME figure out why it doesn't work when Iterator.Element == Equatable
```
Use `Iterator.Element : Equatable`

czechboy0 replied:
Ah, thank you!!!

mxcl commented:
I appreciate the effort, but this wasn't merged earlier because it is really several PRs. The additional tests are most interesting to me, and I'm happy to have a general Collection utils organization. The rest is not really desired, though I certainly can see the improvements, I'd rather improve the tests in general, but I'm waiting on the Linux test generation work.

I understand URL.scheme is not general, but I dislike `repoScheme`, and it's not as though this is public API in Cocoa or anything.

czechboy0 commented:
@mxcl I renamed `repoScheme` back to `scheme`, so those changes are reverted. What's left is the repo scheme parsing refactoring and the new tests and moved manifests. Are those the changes you wanted to keep? If not, please let me know which part exactly you have reservations about.

mxcl commented:
@swift-ci Please test

czechboy0 commented:
@mxcl @swift-ci seems to be in a lazy mood

mxcl commented:
@swift-ci Please test

czechboy0 commented:
Hitting the same WatchKit test failure as everyone else.

czechboy0 commented:
@mxcl Please rerun CI

mxcl commented:
@swift-ci Please test

czechboy0 commented:
\o/
